src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1671,7 +1671,7 @@ private IForLoopOperation CreateBoundForStatementOperation(BoundForStatement bou
                                                                 argumentsToParametersOpt: default,
                                                                 defaultArguments: default,
                                                                 argumentRefKindsOpt: default,
-                                                                expanded: false,
+                                                                expanded: enumeratorMethod.Parameters.Any(p => p.IsParams),
```

Comment:
>enumeratorMethod.Parameters.Any(p => p.IsParams) [](start = 74, length = 48)

This calculation feels fragile and possibly incorrect in general case. Presence of ```params``` modifier doesn't actually imply that the argument always has an expanded form. For example, the parameter might not have the right type. Or ```params``` which is just an attribute in metadata can be on a "wrong" parameter. Etc. See, ```OverloadResolution.IsValidParams``` for example. Rather than trying to re-infer this information, I think we should find a way to thread it through from the binding. Also, with recent work to move creation of default values to binding, I would expect that would also take care of param arrays, when necessary.